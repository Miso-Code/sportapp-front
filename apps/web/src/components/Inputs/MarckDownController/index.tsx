import InputMarckDown from '@/components/InputMarckDown'
import { Props } from './interfaces'
import { Controller, FieldValues } from 'react-hook-form'

export default function MarckDownController<
	T extends FieldValues = FieldValues
>({ control, label, name }: Props<T>) {
	return (
		<Controller
			control={control}
			name={name}
			render={({ field: { value, onChange } }) => (
				<InputMarckDown
					value={value}
					label={label}
					onChange={onChange}
				/>
			)}
		/>
	)
}
