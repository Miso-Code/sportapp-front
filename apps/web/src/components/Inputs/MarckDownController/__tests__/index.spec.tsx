import MarckDownController from '..'
import { render, renderHook, RenderResult } from '@testing-library/react'
import { useForm } from 'react-hook-form'

jest.mock('@uiw/react-md-editor', () => ({
	__esModule: true,
	default: () => <div>MarkdownEditor</div>
}))

describe('MarckDownController', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		const { result } = renderHook(() => useForm())
		wrapper = render(
			<MarckDownController
				control={result.current.control}
				label='Test'
				name='test'
			/>
		)
	})

	it('should render the component', () => {
		expect(wrapper.container).toMatchSnapshot()
	})
})
