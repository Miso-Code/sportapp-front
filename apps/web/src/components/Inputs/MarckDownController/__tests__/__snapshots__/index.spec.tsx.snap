// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarckDownController should render the component 1`] = `
<div>
  <div
    class="w-full"
    data-color-mode="light"
  >
    <label
      class="MuiTypography-root MuiTypography-body1 text-[#00000099] leading-8 css-ahj2mt-MuiTypography-root"
    >
      Test
    </label>
    <div
      class="wmde-markdown-var"
    >
       
    </div>
    <div>
      MarkdownEditor
    </div>
  </div>
</div>
`;
